Snakefile

```diff
@@ -43,7 +43,12 @@ run = config.get("run", {})
 RDIR = run["name"] + "/" if run.get("name") else ""
 CDIR = RDIR if not run.get("shared_cutouts") else ""
 
-load_data_paths = get_load_paths_gegis("data", config)
+if os.path.exists(os.path.join("data", config["load_options"]["ssp"])):
+    load_data_paths = get_load_paths_gegis("data", config, check_existence=True)
+else:
+    # demand profiles are not available yet -> no opportunity to use custom load files
+    load_data_paths = get_load_paths_gegis("data", config, check_existence=False)
```

Comment:
Mmmm I think the check_existence can lead to misleading effects due to when that is evaluated.
I think we should not have this option and keep the previous behavior maybe?

The reason is that the file may not exist when the function is triggered.
For example, with a fresh run, the file does not exists at the beginning because it is moved here from retrieve_databundle.

Maybe the check on whether the file is missing should be moved when the files are actually read, but if the file is missing, the workflow breaks so it may not be needed.
Except the check on whether all selected countries are found, I expected that check to be there already but may be good to crosscheck if you wanted to address that here

Reply:
Thanks for double-checking @davide-f 🙂 This file-existence issue is quite a tricky part, in fact. 

The point is snakemake parses all the input paths before building DAG and starting to execute anything. Which means that `load_data_paths` is evaluated before any of the rule would run, including `retrieve_databundle`. So, it's possible that `load_data_paths` is evaluated when demand files are not loaded yet. It has been perfectly alright before, when we were not interested if the demand files exist. But the demand parser must check which exactly inputs present, and breaks if there is no any inputs yet.

If we would restore the original `get_load_paths_gegis` removing the check-existence condition, that will lead to the error in a fresh run. That has been a reason of CI being unhappy previously, which I have looked to fix. Not sure the current implementation is the most elegant one, but it works and hopefully doesn't introduce breaking changes into the workflow.

Agree that it can also work if we would check if the data folder exist directly in the `load_data_paths` (an example of [a similar solution](https://stackoverflow.com/q/46636844/8465924)). Have implemented and tested the improvement and happy to have your opinion on that 🙂

Reply:
Ahhhh right, good point! I catched the issue but not completely the side effects, great explanation.
I'm not a great fan of that long if case; I got another idea; adding a review comment and let's see what's your idea.
Already this is good :)